xarray/backends/h5netcdf_.py

```diff
@@ -468,7 +468,7 @@ def guess_can_open(self, filename_or_obj: T_PathFileOrDataStore) -> bool:
             return magic_number.startswith(b"\211HDF\r\n\032\n")
 
         if isinstance(filename_or_obj, str | os.PathLike):
-            _, ext = os.path.splitext(filename_or_obj)
```

Comment:
Intentionally not stripping any query params that might be present in dap query so that h5netcdf does not claim to be able to open it, as it's my undersstanding that it cannot